pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -316,6 +316,81 @@ self: super: {
     '';
   });
 
+  coq_nvim = super.coq_nvim.overrideAttrs (old: {
+    passthru.python3Dependencies = let
+
+      # Extract the version of the dependencies from the requirements.txt file
+      requirements = builtins.readFile "${old.src}/requirements.txt";
+      requirementsLines = lib.splitString "\n" requirements;
+
+      extractLine = pname: lib.findSingle
+        (lib.strings.hasPrefix pname)
+        "none"
+        "multiple"
+        requirementsLines;
+
+      extractDepVersion = pname:
+        let
+          line = extractLine pname;
+          hashPart = lib.lists.last (lib.splitString "/" line);
+          hash = lib.strings.removeSuffix ".tar.gz" hashPart;
+        in hash;
+    in ps: with ps; [
+      pynvim
+      pyyaml
+      (let
+        version = extractDepVersion "pynvim_pp";
+      in buildPythonPackage {
+        pname = "pynvim_pp";
+        inherit version;
+        format = "pyproject";
+        propagatedBuildInputs = [ setuptools pynvim ];
+        src = fetchFromGitHub {
+          owner = "ms-jpq";
+          repo = "pynvim_pp";
+          rev = version;
+          fetchSubmodules = false;
+
+          # This maps the current version of pynvim_pp to the sha256 of its source.
+          # Hence, if the dependency version changes in the requirements.txt, the build will fail.
+          hash = { "456765f6dc8cef6df39ae3e61c79585258d38517" = "sha256-edDDzxR60ILFbv1CnYGxW/sJDMeRIFPjmTeiMRdvA3k="; }.${version};
```

Comment:
I am not really sure how I feel about this. This looks much more complicated than it should, and also will unnecessary break almost every time someone upgrades the Vim plugins.

If you want to keep the code in a way to ensure that the code will break after updates, I would suggest something simpler, like:

```nix
assert version == "456765f6dc8cef6df39ae3e61c79585258d38517";
```

Inside the `let in` above. Then the `src` part becomes just:

```nix
        src = fetchFromGitHub {
          owner = "ms-jpq";
          repo = "pynvim_pp";
          rev = version;
          fetchSubmodules = false;
          hash = "sha256-edDDzxR60ILFbv1CnYGxW/sJDMeRIFPjmTeiMRdvA3k=";
        }
```

However, maybe I would just set the known rev/hash that is today, and in case of upgrades we can just let it upgrade until something breaks visibly and someone is interested in upgrading it. Because it is not like every upgrade is breaking, but having your configuration broken because someone else bumped something completely unrelated (another vim plugin) is annoying.

You choice since you're the maintainer.

Reply:
Done !